src/header.rs

```diff
@@ -81,6 +83,8 @@ pub struct HeaderPt1 {
     pub padding: [u8; 7],
 }
 
+const _CONST_CHECK_HEADER_PT1_SIZE: [(); !(mem::size_of::<HeaderPt1>() == 16) as usize] = [];
```

Comment:
I moved the size check of `HeaderPt1` from `sanity_check` as it can be done in a const expression.